src/engine/renderer/tr_shade.cpp

```diff
@@ -104,6 +104,12 @@ static void EnableAvailableFeatures()
 		}
 	}
 
+
+	// Disable features that require deluxe mapping to be enabled.
+	glConfig2.normalMapping = glConfig2.deluxeMapping && glConfig2.normalMapping;
```

Comment:
Is that right? Normal maps seem to be used in other places, for example with dynamic lights and relief mapping.

Reply:
As far as I know, normal mapping and specular mapping aren't doable when there is no known light direction.

Right now `r_deluxeMapping` seems to be meant to disable all kinds of deluxe mapping, being from actual deluxe maps or emulation from light grid (or a dummy fallback one). I see no reason to keep this enabled only with dynamic lights, it would even look weird to see the geometry popping only when there is a dynamic light touching the surface (First Halo game did that and this effect felt weird).